lending/loan_management/doctype/loan_type/loan_type.json

```diff
@@ -1,333 +1,329 @@
 {
-"actions": [],
-"autoname": "field:loan_name",
-"creation": "2019-08-29 18:08:38.159726",
-"doctype": "DocType",
-"editable_grid": 1,
-"engine": "InnoDB",
-"field_order": [
-    "loan_name",
-    "maximum_loan_amount",
-    "rate_of_interest",
-    "penalty_interest_rate",
-    "grace_period_in_days",
-    "write_off_amount",
-    "column_break_2",
-    "company",
```

Comment:
I understand; we missed to take care of the formatting in https://github.com/frappe/lending/pull/1.

Reply:
In a way that‘s true. On the other hand, we – you, me, everyone – shouldn‘t have to take care of the JSON formatting every time we‘re editing a DocType in Frappe and therefore reformatting the JSON file.

IMO, we need to do the follwing:
1.) settle with an exactly predictable JSON formatting and indentation style.
2.) make sure Frappe saves the JSON exactly this way.
3.) reformat all JSON files (in Frappe, ERPNext, Lending…) once.
Only then we will never ever be bothered by this problem again!

Reply:
I think 1 and 2 are already there, and yes, we can do 3. I'm not sure why/how #1 changed the JSON formatting of many doctypes, but yes, we missed to notice or take care of it. Also, I'm curious, when you modified the loan type doctype [here](https://github.com/frappe/lending/pull/20/commits/f977fc8c5de7050517769bd75d9689277e0c4017), it should have formatted the whole file to the actual expected JSON format, but it didn't. I wanted to know if you manually edited the JSON file since when I changed the doctype in this PR, it did format the whole file.

Reply:
Not so sure about 1 being intentionally designed.

Frappe currently formats JSON files with a single space indentation per level which is quite hard to see / read and is pretty much uncommon and therefore not even supported by all structured text editors. I would really recommend settling with 2 spaces which is one of the common seni-standards.

Also, while ordering the field properties alphabetically does improve consistency, there would be a point in an order that is easier to understand, such as "label" > "fieldname" > "fieldtype" > "options" > "description" >  "default" > "unique" > "read_only" > "depends_on" > "reqd" > "mandatory_depends_on" and then all the others...

Further discussing this however is way out of scope here. I might open a ticket with frappe/frappe.

Re https://github.com/frappe/lending/pull/20/commits/f977fc8c5de7050517769bd75d9689277e0c4017: Yes, I did manually edit the JSON to avoid all the noise with reformatting. ;)

Reply:
Not so sure about 1 being intentionally designed.

Frappe currently formats JSON files with a single space indentation per level which is quite hard to see / read and is pretty much uncommon and therefore not even supported by all structured text editors. I would really recommend settling with 2 spaces which is one of the common semi-standards.

Also, while ordering the field properties alphabetically does improve consistency, there would be a point in an order that is easier to understand, such as "label" > "fieldname" > "fieldtype" > "options" > "description" >  "default" > "unique" > "read_only" > "depends_on" > "reqd" > "mandatory_depends_on" and then all the others...

Further discussing this however is way out of scope here. I might open a ticket with frappe/frappe.

Re https://github.com/frappe/lending/pull/20/commits/f977fc8c5de7050517769bd75d9689277e0c4017: Yes, I did manually edit the JSON to avoid all the noise with reformatting. ;)